src/main/java/org/currencygoldexchangeapp/services/CurrencyExchangeCalculateService.java

```diff
@@ -0,0 +1,58 @@
+package org.currencygoldexchangeapp.services;
+
+import org.currencygoldexchangeapp.datamodels.CurrencyExchange;
+import org.currencygoldexchangeapp.handlers.ExchangeRateAPIHandler;
+
+import java.math.BigDecimal;
+import java.math.RoundingMode;
+
+public class CurrencyExchangeCalculateService {
+    private final ExchangeRateAPIHandler exchangeRateAPIHandler;
+
+    public CurrencyExchangeCalculateService(ExchangeRateAPIHandler exchangeRateAPIHandler) {
+        this.exchangeRateAPIHandler = exchangeRateAPIHandler;
+    }
+
+    public CurrencyExchange calculateExchangeAmount (String sourceCurrency, double amount, String targetCurrency, String date) {
```

Comment:
About the naming here:  what does exchange mean in this context?

Reply:
The name of the calculateExchangeAmount method suggests that this method deals with calculations related to the exchange of one currency for another.

In my opinion, the name of this method is quite clear, but I could make it even more precise e.g. calculateAmountInTargetCurrency